source/predicates/predicate.ts

```diff
@@ -161,7 +161,7 @@ export class Predicate<T = unknown> implements BasePredicate<T> {
 	Invert the following validators.
 	*/
 	get not(): this {
-		return not(this);
+		return not(this) as this;
```

Comment:
Do you really need to use `as`?

Reply:
If not using `as`, following error occurred.
```
Type 'Predicate<unknown>' is not assignable to type 'this'.
  'this' could be instantiated with an arbitrary type which could be unrelated to 'Predicate<unknown>'.ts(2322)
```
<img width="784" alt="スクリーンショット 2021-01-27 8 33 00" src="https://user-images.githubusercontent.com/50798936/105920199-4a137d80-607a-11eb-9400-e631dad304c7.png">

It is because`not` function return `Predicate` type now.

```ts:not.ts
...
export const not = (predicate: Predicate): Predicate => {
...
```

If using `any` at `not.ts`, I don't need to use `as` at `predicate.ts`

```ts:not.ts
...
export const not = (predicate: Predicate): any => {
...
```